engine-rest bundles an incompatible Jackson 3 datatype (`jackson-datatype-jsr310:3.0.0-rc2`)

### Symptom
Observed with the agentic plugins: Spring AI fails to parse the Ollama chat response's `created_at` (`java.time.Instant`) with a `NoSuchFieldError`, and the orchestration job fails repeatedly. The stock engine itself shows no symptom (see Root cause).

The error is thrown in spring-web's generic JSON message converter (`AbstractJacksonHttpMessageConverter`, see trace below), not in Spring AI code — so any other code in the same JVM that uses Spring's HTTP message converters with `java.time` types (for example a `JavaDelegate` calling an external API via `RestClient`/`RestTemplate`) is expected to fail the same way, though this has not been separately tested.

```
WARN  o.f.f.b.engine.jobexecutor : ENGINE-14006 Exception while executing job ...
java.lang.RuntimeException: Retry policy for operation 'org.springframework.ai.ollama.OllamaChatModel' exhausted; aborting execution
	at org.springframework.ai.ollama.OllamaChatModel.internalCall(OllamaChatModel.java:234)
	at ...advisor.ToolCallingAdvisor.adviseCall(ToolCallingAdvisor.java:150)
	at ...DefaultChatClient$DefaultCallResponseSpec.chatResponse(DefaultChatClient.java:626)
	at org.finos.fluxnova.bpm.engine.ai.agent.llm.service.SpringAiLlmService.call(...)
Caused by: java.lang.NoSuchFieldError: Class tools.jackson.databind.DeserializationFeature
    does not have member field 'tools.jackson.databind.DeserializationFeature ADJUST_DATES_TO_CONTEXT_TIME_ZONE'
	at tools.jackson.datatype.jsr310.deser.InstantDeserializer.shouldAdjustToContextTimezone(InstantDeserializer.java:344) ~[jackson-datatype-jsr310-3.0.0-rc2.jar:3.0.0-rc2]
	at tools.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:325) ~[jackson-datatype-jsr310-3.0.0-rc2.jar:3.0.0-rc2]
	at tools.jackson.databind.deser.bean.BeanDeserializer.deserialize(BeanDeserializer.java:200) ~[jackson-databind-3.1.0.jar:3.1.0]
	at tools.jackson.databind.ObjectReader.readValue(ObjectReader.java:1270) ~[jackson-databind-3.1.0.jar:3.1.0]
	at org.springframework.http.converter.AbstractJacksonHttpMessageConverter.readJavaType(...) ~[spring-web-7.0.6.jar]
	at org.springframework.ai.ollama.api.OllamaApi.chat(OllamaApi.java:119) ~[spring-ai-ollama-2.0.0.jar]
```

### Root cause
`engine-rest` pins `tools.jackson.datatype:jackson-datatype-jsr310` to **`3.0.0-rc2`** while the rest of Jackson 3 resolves to **`3.1.x`** (`jackson-databind 3.1.0` shown above). That RC's `InstantDeserializer` references `DeserializationFeature.ADJUST_DATES_TO_CONTEXT_TIME_ZONE`, a field removed in databind 3.1.x → `NoSuchFieldError`.

- There is no matching release: `tools.jackson.datatype:jackson-datatype-jsr310` only has `3.0.0-rc1/rc2` on Maven Central, while databind is at `3.1.x`.
- The dependency is **unused**: `JacksonConfigurator` uses Jackson 3 databind's built-in `java.time` support (`tools.jackson.databind.cfg.DateTimeFeature`) and does not register `JavaTimeModule`.
- The broken module only affects mappers built **with module discovery** (ServiceLoader-based: `MapperBuilder.findModules(...)`/`addModules(...)`, or the equivalent `findAndAddModules()` shorthand). Spring Framework 7's `AbstractJacksonHttpMessageConverter` does this in its `initModules()` when building its default mapper, which is how the module gets registered on the spring-web path. Once registered, it fails in both directions: deserialization references the removed `DeserializationFeature.ADJUST_DATES_TO_CONTEXT_TIME_ZONE`, serialization references the removed `SerializationFeature.WRITE_DATES_AS_TIMESTAMPS` (both moved to `DateTimeFeature` in databind 3.1). Mappers built without discovery are unaffected even with the jar present — databind's built-in `java.time` support handles those types.
- The stock engine shows no symptom and the defect is not triggerable via the engine REST API: the engine REST mapper is built without module discovery, and no engine REST DTO deserializes `java.time` (all `java.util.Date`; Spin uses Jackson 2). The failure only occurs in integration code running in the same JVM.
- Introduced by the Spring Boot 4 / Spring AI 2.0 upgrade; the jsr310 dep (originally added for Jackson-2 `JavaTimeModule`) was carried over and pinned to the only available Jackson-3 build.

### Fix
Remove the dependency from both engine-rest poms — Jackson 3 databind handles `java.time` natively.

```diff
--- a/engine-rest/engine-rest/pom.xml
+++ b/engine-rest/engine-rest/pom.xml
@@
-    <dependency>
-      <groupId>tools.jackson.datatype</groupId>
-      <artifactId>jackson-datatype-jsr310</artifactId>
-      <version>3.0.0-rc2</version>
-    </dependency>
-
     <!-- Jakarta XML Bind API - needed for Jackson JAXB module -->
```
```diff
--- a/engine-rest/engine-rest-jakarta/pom.xml
+++ b/engine-rest/engine-rest-jakarta/pom.xml
@@
-    <dependency>
-      <groupId>tools.jackson.datatype</groupId>
-      <artifactId>jackson-datatype-jsr310</artifactId>
-      <version>3.0.0-rc2</version>
-    </dependency>
-
     <!-- provided deps -->
```

### Verification
- `engine-rest` `Java8DateTimeTest` (LocalDate → ISO string over REST) still passes.
- Rebuilt Run distro no longer bundles `jackson-datatype-jsr310-3.0.0-rc2`; the agentic Ollama loop deserializes the response and completes.

--- a/engine-rest/engine-rest-jakarta/pom.xml
+++ b/engine-rest/engine-rest-jakarta/pom.xml
@@ -92,12 +92,6 @@
       <version>${version.jackson}</version>
     </dependency>
 
-    <dependency>
-      <groupId>tools.jackson.datatype</groupId>
-      <artifactId>jackson-datatype-jsr310</artifactId>
-      <version>3.0.0-rc2</version>
-    </dependency>
-
     <!-- provided deps -->
     <dependency>
       <groupId>jakarta.servlet</groupId>

--- a/engine-rest/engine-rest/pom.xml
+++ b/engine-rest/engine-rest/pom.xml
@@ -96,12 +96,6 @@
       <scope>test</scope>
     </dependency>
 
-    <dependency>
-      <groupId>tools.jackson.datatype</groupId>
-      <artifactId>jackson-datatype-jsr310</artifactId>
-      <version>3.0.0-rc2</version>
-    </dependency>
-
     <!-- Jakarta XML Bind API - needed for Jackson JAXB module -->
     <dependency>
       <groupId>jakarta.xml.bind</groupId>

